src/calculus/HCAT.jl

```diff
@@ -281,40 +281,38 @@ function size(H::HCAT)
 		eltype(s) <: Int ? push!(size_in,s) : push!(size_in,s...) 
 	end
 	p = vcat([[idx... ] for idx in H.idxs]...)
-	ipermute!(size_in,p)
+	invpermute!(size_in,p)
 
-	size(H.A[1],1), (size_in...)
+	size(H.A[1],1), (size_in...,)
 end
 
-fun_name(L::HCAT) = length(L.A) == 2 ? "["fun_name(L.A[1])*","*fun_name(L.A[2])*"]" : "HCAT"
+fun_name(L::HCAT) = length(L.A) == 2 ? "["*fun_name(L.A[1])*","*fun_name(L.A[2])*"]" : "HCAT"
 
 function domainType(H::HCAT) 
-	domain = vcat([typeof(d)<:Tuple ? [d...] : d  for d in domainType.(H.A)]...)
+    domain = vcat([typeof(d)<:Tuple ? [d...] : d  for d in domainType.(H.A)]...)
 	p = vcat([[idx... ] for idx in H.idxs]...)
-	ipermute!(domain,p)
-	return (domain...)
+	invpermute!(domain,p)
+	return (domain...,)
 end
-codomainType(L::HCAT) = codomainType.(L.A[1])
+codomainType(L::HCAT) = codomainType.(Ref(L.A[1]))
```

Comment:
Why this `Ref`? I'm just curious, I don't now how that works

Reply:
from the [0.7 release notes](https://docs.julialang.org/en/v0.7.0/NEWS/)

> Previously, broadcast defaulted to treating its arguments as scalars if they were not arrays. This behavior is deprecated, and in the future broadcast will default to iterating over all its arguments. Wrap arguments you wish to be treated as scalars with Ref() or a 1-tuple. Package developers can choose to allow a non-iterable type T to always behave as a scalar by implementing broadcastable(x::T) = Ref(x) (#26212).

I'm still digesting this though...